system/lib/compiler-rt/lib/asan/asan_errors.cc

```diff
@@ -467,6 +467,12 @@ ErrorGeneric::ErrorGeneric(u32 tid, uptr pc_, uptr bp_, uptr sp_, uptr addr,
       scariness.Scare(bug_type_score + read_after_free_bonus, bug_descr);
       if (far_from_bounds) scariness.Scare(10, "far-from-bounds");
     }
+#if SANITIZER_EMSCRIPTEN
+    else if (AddrIsInShadow(addr)) {
+      bug_descr = "null-pointer-dereference";
```

Comment:
It isn't necessarily a null pointer, right? It could be some other invalid low pointer. Would it make sense to have a different error message in that case?

Reply:
Well, null pointer dereference does not necessarily result in accesses to address 0. It could just be low addresses when doing `((struct x*) NULL)->y`. But we probably should cap this at like 64 KB instead of treating all shadow as null dereference.